src/app/core/models/content-page-tree-view/content-page-tree-view.model.ts

```diff
@@ -31,7 +31,7 @@ function getContentPageTreeElements(
   // If current content page is part of element subtree or equals the element itself, then all children of elements should be analyzed
   if (tree.edges[elementId] && isContentPagePartOfPageTreeElement(tree, currentContentPageId, elementId)) {
     treeElements = tree.edges[elementId]
-      .map(child => getContentPageTreeElements(tree, child, rootId, currentContentPageId))
+      .map(child => (child === elementId ? [] : getContentPageTreeElements(tree, child, rootId, currentContentPageId)))
```

Comment:
Instead of expanding the `map` I would introduce a `filter(child => child !== elementId)` before.
That should do the same thing, right?

Reply:
Filter would return undefined.
I think that I had encounter a side effect if the element was not set to an empty array

Reply:
I have added a test case, and youwere right, the filter is enought.